sympy/polys/polytools.py

```diff
@@ -3816,6 +3817,89 @@ def nth_power_roots_poly(f, n):
 
         return r.replace(t, x)
 
+    def which_roots(f, candidates, real=True):
+        """
+        Given a superset of roots of f, finds which ones are
+        roots of f. Note this won't work properly if candidates is
+        not a superset of the roots of f. The list returned has
+        unique elements, is at most the length of the number of
+        unique roots of f, and preserves the order of the original
+        list of candidates. The `real` argument only controls
+        the root counting: if `real=True`, the `.count_roots()` method
+        is used (only supported over ZZ, QQ, or AlgebraicField); if
+        `real=False`, the degree is used.
+
+        Examples
+        ========
+
+        >>> from sympy import Poly, I, sqrt
+        >>> from sympy.abc import x
+
+        >>> f = Poly(x**4 - 1)
+
+        >>> f.which_roots([-1, 1, 0, -2, 2], real=True)
+        [-1, 1]
+        >>> f.which_roots([-1, 1, -I, I, 0], real=False)
+        [-1, 1, -I, I]
+        >>> f.which_roots([-1, 1, 1, 1, 1], real=True)
+        [-1, 1]
+        >>> f.which_roots([-1, -1, 1, 1], real=False)
+        [-1, 1]
+
+        # expected behaviour if not a superset of roots
+        >>> f.which_roots([2, 5], real=True)
+        [2, 5]
+
+        # lifting to rational coeffs produces extraneous roots
+        # which we can filter out with this method
+        >>> f = Poly(sqrt(2)*x**3 + x**2 - 1, x, extension=True)
+        >>> f.lift()
```

Comment:
The lines that look like comments here should not be like comments. This is prose in the docs. We don't have comments in the docs.

Reply:
Changed to prose!